File: src/main/java/com/hjj/apiserver/service/MainService.java
package com.hjj.apiserver.service;

import com.hjj.apiserver.dto.MainDto;
import com.hjj.apiserver.repositroy.DepositRepository;
import com.hjj.apiserver.repositroy.SavingRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class MainService {

    private final DepositRepository depositRepository;
    private final SavingRepository savingRepository;


    public MainDto.ResponseMain findMain(){

        return MainDto.ResponseMain.builder()
                .deposits(depositRepository.findDepositByHome())
                .savings(savingRepository.findSavingByHome())
                .build();
    }
}

File: src/main/kotlin/com/hjj/apiserver/service/MainService.kt
package com.hjj.apiserver.service

import com.hjj.apiserver.dto.main.response.MainFindResponse
import com.hjj.apiserver.repository.deposit.DepositRepository
import com.hjj.apiserver.repository.saving.SavingRepository
import org.springframework.stereotype.Service

@Service
class MainService(
    private val depositRepository: DepositRepository,
    private val savingRepository: SavingRepository,
) {

    fun findMain():MainFindResponse{
        return MainFindResponse(
            deposits = depositRepository.findDepositByHome(),
            savings = savingRepository.findSavingByHome(),
        )
    }
}